lib/identity_cache/configuration_dsl.rb

```diff
@@ -42,18 +40,19 @@ module ClassMethods
       # * unique: if the index would only have unique values
       #
       def cache_index(*fields)
-        ensure_base_model
         raise NotImplementedError, "Cache indexes need an enabled primary index" unless primary_cache_index_enabled
         options = fields.extract_options!
-        self.cache_indexes.push fields
+        unique = options[:unique] || false
+        cache_attribute primary_key, by: fields, unique: unique
```

Comment:
@camilo this is the line that causes problems for Shopify's container build, which doesn't allow schema queries, since it happens during initialization.